exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/SqlPatternContainsMatcher.java

```diff
@@ -19,44 +19,283 @@
+    @Override
+    protected final int match(int start, int end, DrillBuf drillBuf) {
+      final int lengthToProcess = end - start;
+      final byte firstPattByte  = patternArray[0];
 
-    // no match if input string length is less than pattern length
-    if (txtLength < patternLength) {
+      // simplePattern string has meta characters i.e % and _ and escape characters removed.
+      // so, we can just directly compare.
+      for (int idx = 0; idx < lengthToProcess; idx++) {
+        byte inputByte = drillBuf.getByte(start + idx);
+
+        if (firstPattByte != inputByte) {
+          continue;
+        }
+        return 1;
+      }
       return 0;
     }
+  }
 
+  /** Handles patterns with length two */
+  private final class Matcher2 extends MatcherFcn {
 
-    final int outerEnd = txtLength - patternLength;
+    private Matcher2() {
+      super();
+    }
 
-    outer:
-    for (int txtIndex = 0; txtIndex <= outerEnd; txtIndex++) {
+    /** {@inheritDoc} */
+    @Override
+    protected final int match(int start, int end, DrillBuf drillBuf) {
+      final int lengthToProcess = end - start - 1;
+      final byte firstPattByte  = patternArray[0];
+      final byte secondPattByte = patternArray[1];
 
       // simplePattern string has meta characters i.e % and _ and escape characters removed.
       // so, we can just directly compare.
-      for (int patternIndex = 0; patternIndex < patternLength; patternIndex++) {
-        if (patternByteBuffer.get(patternIndex) != drillBuf.getByte(start + txtIndex + patternIndex)) {
-          continue outer;
+      for (int idx = 0; idx < lengthToProcess; idx++) {
+        final byte firstInByte = drillBuf.getByte(start + idx);
+
+        if (firstPattByte != firstInByte) {
+          continue;
+        } else {
+          final byte secondInByte = drillBuf.getByte(start + idx +1);
+
+          if (secondInByte == secondPattByte) {
+            return 1;
+          }
         }
       }
+      return 0;
+    }
+  }
+
+  /** Handles patterns with length three */
+  private final class Matcher3 extends MatcherFcn {
 
-      return 1;
+    private Matcher3() {
+      super();
     }
 
-    return  0;
+    /** {@inheritDoc} */
+    @Override
+    protected final int match(int start, int end, DrillBuf drillBuf) {
+      final int lengthToProcess  = end - start -2;
+      final byte firstPattByte   = patternArray[0];
+      final byte secondPattByte  = patternArray[1];
+      final byte thirdPattByte   = patternArray[2];
+
+      // simplePattern string has meta characters i.e % and _ and escape characters removed.
+      // so, we can just directly compare.
+      for (int idx = 0; idx < lengthToProcess; idx++) {
+        final byte inputByte = drillBuf.getByte(start + idx);
+
+        if (firstPattByte != inputByte) {
+          continue;
+        } else {
+          final byte secondInByte = drillBuf.getByte(start + idx +1);
+          final byte thirdInByte  = drillBuf.getByte(start + idx +2);
```

Comment:
Is it faster to say `secondInByte = thirdInByte`; shuffle values between variables in each iteration rather than diving into direct memory multiple types for the same byte?